weblate/trans/tests/test_reports.py

```diff
@@ -167,7 +167,11 @@ def test_credits_view_html(self) -> None:
     def test_credits_blank_period(self):
         period = ""
         response = self.get_credits("json", period=period, follow=True)
-        self.assertContains(response, "Error in parameter period: Invalid date!")
+        self.assertEqual(response.status_code, 200)
+        self.assertJSONEqual(
+            response.content.decode(),
+            [{"Czech": [["weblate@example.org", "Weblate <b>Test</b>", 1]]}],
+        )
```

Comment:
Reports should not be allowed without a period for performance reasons.

Reply:
Made the `DateRangeField` in the `ReportsForm` required. I also adjusted the `DateRangeField` to disallow empty values when `required` field is set to True.